CMakeLists.txt

```diff
@@ -50,82 +50,95 @@ foreach (submodule ${submodules})
 endforeach()
 
 
+#Create configure options
+option(DEBUG "Build with debugging symbols" OFF) #replace with CMAKE_BUILD_TYPE?
+option(OPTIMIZE " Build with optimizations" ON) #replace with CMAKE_BUILD_TYPE?
+option(GCOV "Build with gcov profiling support" OFF)
+option(CLANGCOV "Build with clangcov profiling support" OFF)
+set(RANDOMIZATION "1" CACHE STRING "0: no randomization. 1: freelist init only.  2: freelist init + free fastpath")
+set_property(CACHE RANDOMIZATION PROPERTY STRINGS "0;1;2")
+option(DISABLE_MESHING "Disable meshing" OFF)
+option(SUFFIX "Always suffix the mesh library with randomization + meshing info" OFF)
+option(CLANG "Build with clang" OFF)
+
+
+
+
+#Parse options
+set(CXX_FLAGS )
+set(ENV{CXX_FLAGS} )
+
+if (${CLANG})
+    set(CMAKE_CXX_COMPILER clang++)
+    set(CMAKE_C_COMPILER clang)
+endif()
+
 #Identify compiler and set specific flag
-if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU")
     set(GCC TRUE)
-    set(CLANG FALSE)
+    set(CLANGC FALSE)
     set(MSVC FALSE)
     set(ICC FALSE)
 endif()
 
-if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
     set(GCC FALSE)
-    set(CLANG TRUE)
+    set(CLANGC TRUE)
     set(MSVC FALSE)
     set(ICC FALSE)
 endif()
 
-if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
+if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "MSVC")
     set(GCC FALSE)
-    set(CLANG FALSE)
+    set(CLANGC FALSE)
     set(MSVC TRUE)
     set(ICC FALSE)
     set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
     set(BUILD_SHARED_LIBS TRUE)
 endif()
 
-if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
+if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Intel")
```

Comment:
are you actually testing w/ ICC?

Reply:
Not yet. Muscle memory from when I was working with Parallel Studio.